app/models/registry.rb

```diff
@@ -105,19 +105,47 @@ def reachable?
 
   protected
 
+  # Fetch the tag being pushed through the given target object.
+  def get_tag_from_target(target)
+    case target["mediaType"]
+    when "application/vnd.docker.distribution.manifest.v2+json",
+      "application/vnd.docker.distribution.manifest.list.v2+json"
+      get_tag_from_list(target["repository"])
+    else
```

Comment:
Just an idea, can't we make a second case for manifest v2 schema 1 and then have the `else` block log an error. By doing that we would immediately notice when manifest v2 schema 3 is released.